<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Getting Start with Assisted CIDER (asCIDER) • CIDER</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Getting Start with Assisted CIDER (asCIDER)">
<meta name="description" content="Hu et al. An interpretable meta-clustering framework for single-cell RNA-Seq data integration and evaluation">
<meta property="og:description" content="Hu et al. An interpretable meta-clustering framework for single-cell RNA-Seq data integration and evaluation">
<meta property="og:image" content="https://zhiyuan-hu-lab.github.io/CIDER/reference/figures/card.png">
<!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-207749729-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-207749729-1');
</script>
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">CIDER</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.2</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/asCIDER.html">Getting Start with Assisted CIDER (asCIDER)</a></li>
    <li><a class="dropdown-item" href="../articles/dnCIDER_highlevel.html">Getting Start with De Novo CIDER (dnCIDER): Cross-Species Pancreas Integration</a></li>
    <li><a class="dropdown-item" href="../articles/dnCIDER.html">Detailed walk-through of de novo CIDER (dnCIDER) on pancreas data</a></li>
    <li><a class="dropdown-item" href="../articles/evaluation.html">Evaluating scRNA-seq Integration Quality with CIDER</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/zhiyuan-hu-lab/CIDER/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Getting Start with Assisted CIDER (asCIDER)</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/zhiyuan-hu-lab/CIDER/blob/HEAD/vignettes/asCIDER.Rmd" class="external-link"><code>vignettes/asCIDER.Rmd</code></a></small>
      <div class="d-none name"><code>asCIDER.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>This vignette performs asCIDER, a meta-clustering method, on a
cross-species pancreas dataset. AsCIDER is aimed to achieve the
clustering task in data confounded by unwanted variables. In this
scenario, the unwanted variable is species effects.</p>
<p>asCIDER is short for assisted CIDER, assisted by the prior
batch-specific annotations or clustering results.</p>
</div>
<div class="section level2">
<h2 id="set-up">Set up<a class="anchor" aria-label="anchor" href="#set-up"></a>
</h2>
<p>In addition to <strong>CIDER</strong>, we will load the following
packages:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/zhiyuan-hu-lab/CIDER" class="external-link">CIDER</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">pheatmap</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://wilkelab.org/cowplot/" class="external-link">cowplot</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://sjmgarnier.github.io/viridis/" class="external-link">viridis</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="pancreas-data">Pancreas data<a class="anchor" aria-label="anchor" href="#pancreas-data"></a>
</h2>
<p>The example data can be downloaded from <a href="https://figshare.com/s/d5474749ca8c711cc205" class="external-link uri">https://figshare.com/s/d5474749ca8c711cc205</a>.</p>
<p>This
data<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi></mi><mn>1</mn></msup><annotation encoding="application/x-tex">^1</annotation></semantics></math>
contain cells from human (8241 cells) and mouse (1886 cells).</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="st">"../data/pancreas_counts.RData"</span><span class="op">)</span> <span class="co"># count matrix</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="st">"../data/pancreas_meta.RData"</span><span class="op">)</span> <span class="co"># meta data/cell information</span></span>
<span><span class="va">seu</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://mojaveazure.github.io/seurat-object/reference/CreateSeuratObject.html" class="external-link">CreateSeuratObject</a></span><span class="op">(</span>counts <span class="op">=</span> <span class="va">pancreas_counts</span>, meta.data <span class="op">=</span> <span class="va">pancreas_meta</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">seu</span><span class="op">$</span><span class="va">Batch</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; human mouse </span></span>
<span><span class="co">#&gt;  8241  1886</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="exam-if-data-are-confounded">Exam if data are confounded<a class="anchor" aria-label="anchor" href="#exam-if-data-are-confounded"></a>
</h2>
<p>Prior to use CIDER (or other integration methods for clustering) it
is important to exam if clustering will be confounded by the
cross-species factors.</p>
<p>Her we first perform the conventional
Seurat<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi></mi><mn>2</mn></msup><annotation encoding="application/x-tex">^2</annotation></semantics></math>
clustering pipeline.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">seu</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/NormalizeData.html" class="external-link">NormalizeData</a></span><span class="op">(</span><span class="va">seu</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">seu</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindVariableFeatures.html" class="external-link">FindVariableFeatures</a></span><span class="op">(</span><span class="va">seu</span>, selection.method <span class="op">=</span> <span class="st">"vst"</span>, nfeatures <span class="op">=</span> <span class="fl">2000</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">seu</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/ScaleData.html" class="external-link">ScaleData</a></span><span class="op">(</span><span class="va">seu</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">seu</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/RunPCA.html" class="external-link">RunPCA</a></span><span class="op">(</span><span class="va">seu</span>, npcs <span class="op">=</span> <span class="fl">20</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">seu</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/RunTSNE.html" class="external-link">RunTSNE</a></span><span class="op">(</span><span class="va">seu</span>, reduction <span class="op">=</span> <span class="st">"pca"</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">12</span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="confounded-dimension-reduction">Confounded dimension reduction<a class="anchor" aria-label="anchor" href="#confounded-dimension-reduction"></a>
</h3>
<p>The PCA and <em>t</em>-SNE plots both showed that the data are
confounded by species.</p>
<p>The <code>scatterPlot</code> function is used to generate dimension
reduction figures. It takes input of a Seurat object (<code>seu</code>
here), the name of reduction (<code>pca</code> and <code>tsne</code>
here), the variable deciding dot colours (<code>Batch</code> here) and
the title of plots. See more information by
<code><a href="../reference/scatterPlot.html">?scatterPlot</a></code></p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/scatterPlot.html">scatterPlot</a></span><span class="op">(</span><span class="va">seu</span>, <span class="st">"pca"</span>,colour.by <span class="op">=</span> <span class="st">"Batch"</span>, title <span class="op">=</span> <span class="st">"PCA"</span><span class="op">)</span> </span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/scatterPlot.html">scatterPlot</a></span><span class="op">(</span><span class="va">seu</span>, <span class="st">"tsne"</span>,colour.by <span class="op">=</span> <span class="st">"Batch"</span>, title <span class="op">=</span> <span class="st">"t-SNE"</span><span class="op">)</span> </span>
<span><span class="fu"><a href="https://wilkelab.org/cowplot/reference/plot_grid.html" class="external-link">plot_grid</a></span><span class="op">(</span><span class="va">p1</span>, <span class="va">p2</span><span class="op">)</span></span></code></pre></div>
<p><img src="asCIDER_files/figure-html/scatterPlot-confounded-1.png" width="672"></p>
</div>
<div class="section level3">
<h3 id="confounded-clustering-results">Confounded clustering results<a class="anchor" aria-label="anchor" href="#confounded-clustering-results"></a>
</h3>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">seu</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindNeighbors.html" class="external-link">FindNeighbors</a></span><span class="op">(</span><span class="va">seu</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">12</span><span class="op">)</span></span>
<span><span class="co">#&gt; Computing nearest neighbor graph</span></span>
<span><span class="co">#&gt; Computing SNN</span></span>
<span><span class="va">seu</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindClusters.html" class="external-link">FindClusters</a></span><span class="op">(</span><span class="va">seu</span><span class="op">)</span></span>
<span><span class="co">#&gt; Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Number of nodes: 10127</span></span>
<span><span class="co">#&gt; Number of edges: 352494</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Running Louvain algorithm...</span></span>
<span><span class="co">#&gt; Maximum modularity in 10 random starts: 0.9149</span></span>
<span><span class="co">#&gt; Number of communities: 22</span></span>
<span><span class="co">#&gt; Elapsed time: 1 seconds</span></span>
<span></span>
<span><span class="fu"><a href="../reference/scatterPlot.html">scatterPlot</a></span><span class="op">(</span><span class="va">seu</span>, <span class="st">"tsne"</span>,colour.by <span class="op">=</span> <span class="st">"seurat_clusters"</span>, title <span class="op">=</span> <span class="st">"t-SNE"</span><span class="op">)</span> </span></code></pre></div>
<p><img src="asCIDER_files/figure-html/unnamed-chunk-3-1.png" width="480"></p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">seu</span><span class="op">$</span><span class="va">seurat_clusters</span>, <span class="va">seu</span><span class="op">$</span><span class="va">Batch</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">res</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>fill<span class="op">=</span><span class="va">Var2</span>, y<span class="op">=</span><span class="va">Freq</span>, x<span class="op">=</span><span class="va">Var1</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_bar</a></span><span class="op">(</span>position<span class="op">=</span><span class="st">"stack"</span>, stat<span class="op">=</span><span class="st">"identity"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">"CIDER_cluster"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">"Proportions"</span><span class="op">)</span></span></code></pre></div>
<p><img src="asCIDER_files/figure-html/unnamed-chunk-4-1.png" width="576"></p>
</div>
</div>
<div class="section level2">
<h2 id="ascider">asCIDER<a class="anchor" aria-label="anchor" href="#ascider"></a>
</h2>
<div class="section level3">
<h3 id="prepare-initial-clusters">Prepare initial clusters<a class="anchor" aria-label="anchor" href="#prepare-initial-clusters"></a>
</h3>
<p>asCIDER uses existing within-batch clustering results. Here we
concatenate the batch ID and the within-batch cluster ID to obtain
cluster-specific groups (i.e. initial clusters).</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">seu</span><span class="op">$</span><span class="va">initial_cluster</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">seu</span><span class="op">$</span><span class="va">Group</span>, <span class="va">seu</span><span class="op">$</span><span class="va">Batch</span>, sep <span class="op">=</span> <span class="st">"_"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">seu</span><span class="op">$</span><span class="va">initial_cluster</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;             acinar_human activated_stellate_human activated_stellate_mouse </span></span>
<span><span class="co">#&gt;                      932                      275                       14 </span></span>
<span><span class="co">#&gt;              alpha_human              alpha_mouse             b_cell_mouse </span></span>
<span><span class="co">#&gt;                     2241                      191                       10 </span></span>
<span><span class="co">#&gt;               beta_human               beta_mouse              delta_human </span></span>
<span><span class="co">#&gt;                     2455                      894                      592 </span></span>
<span><span class="co">#&gt;              delta_mouse             ductal_human             ductal_mouse </span></span>
<span><span class="co">#&gt;                      218                     1033                      275 </span></span>
<span><span class="co">#&gt;        endothelial_human        endothelial_mouse            epsilon_human </span></span>
<span><span class="co">#&gt;                      214                      139                       17 </span></span>
<span><span class="co">#&gt;              gamma_human              gamma_mouse       immune_other_mouse </span></span>
<span><span class="co">#&gt;                      241                       41                        8 </span></span>
<span><span class="co">#&gt;         macrophage_human         macrophage_mouse               mast_human </span></span>
<span><span class="co">#&gt;                       39                       36                       25 </span></span>
<span><span class="co">#&gt; quiescent_stellate_human quiescent_stellate_mouse            schwann_human </span></span>
<span><span class="co">#&gt;                      160                       47                       10 </span></span>
<span><span class="co">#&gt;            schwann_mouse             t_cell_human             t_cell_mouse </span></span>
<span><span class="co">#&gt;                        6                        7                        7</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="calculate-of-ider-similarity-matrix">Calculate of IDER similarity matrix<a class="anchor" aria-label="anchor" href="#calculate-of-ider-similarity-matrix"></a>
</h3>
<p>The function <code>getIDEr</code> calculate the IDER-based distance
matrix.</p>
<p>By default, it will use the column called “initial_cluster” as
initial clusters, and “Batch” as batch. If you are using columns other
than this two, please revise these two parameters.</p>
<p>For this step, you can choose to use parallel computation by setting
<code>use.parallel = TRUE</code> or not (default
<code>use.parallel = FALSE</code>). The default number of cores used for
parallel computation is
<code>detectCores(logical = FALSE) - 1</code>.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ider</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getIDEr.html">getIDEr</a></span><span class="op">(</span><span class="va">seu</span>, </span>
<span>                group.by.var <span class="op">=</span> <span class="st">"initial_cluster"</span>,</span>
<span>                batch.by.var <span class="op">=</span> <span class="st">"Batch"</span>,</span>
<span>                downsampling.size <span class="op">=</span> <span class="fl">35</span>, </span>
<span>                use.parallel <span class="op">=</span> <span class="cn">FALSE</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="visualise-the-similarity-matrix">Visualise the similarity matrix<a class="anchor" aria-label="anchor" href="#visualise-the-similarity-matrix"></a>
</h3>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">groups</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"alpha"</span>,<span class="st">"beta"</span>,<span class="st">"delta"</span>, <span class="st">"gamma"</span>,<span class="st">"ductal"</span>,<span class="st">"endothelial"</span>, <span class="st">"activated_stellate"</span>, <span class="st">"quiescent_stellate"</span>, <span class="st">"macrophage"</span><span class="op">)</span></span>
<span><span class="va">idx1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">groups</span>, <span class="st">"_human"</span><span class="op">)</span></span>
<span><span class="va">idx2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">groups</span>, <span class="st">"_mouse"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">pheatmap</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/pheatmap/man/pheatmap.html" class="external-link">pheatmap</a></span><span class="op">(</span></span>
<span>  <span class="va">ider</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="va">idx1</span>, <span class="va">idx2</span><span class="op">]</span>,</span>
<span>  color <span class="op">=</span> <span class="fu"><a href="https://sjmgarnier.github.io/viridisLite/reference/viridis.html" class="external-link">inferno</a></span><span class="op">(</span><span class="fl">10</span><span class="op">)</span>,</span>
<span>  border_color <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>  display_numbers <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  cluster_rows <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  cluster_cols <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  width <span class="op">=</span> <span class="fl">7</span>,</span>
<span>  height <span class="op">=</span> <span class="fl">5</span>,</span>
<span>  cellwidth <span class="op">=</span> <span class="fl">22</span>,</span>
<span>  cellheight <span class="op">=</span> <span class="fl">22</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="asCIDER_files/figure-html/heatmap-1.png" width="672"></p>
</div>
<div class="section level3">
<h3 id="perform-final-clustering">Perform final Clustering<a class="anchor" aria-label="anchor" href="#perform-final-clustering"></a>
</h3>
<p>Next we put both Seurat object and the similarity matrix list into
<code>finalClustering</code>. You can either cut trees by height (set
<code>cutree.by = 'h</code>) or by k (set <code>cutree.by = 'k</code>).
The default is cutting by the height of 0.45.</p>
<p>The final clustering results are stored in Colume
<code>cider_final_cluster</code> of Seurat object metadata and can be
extracted using <code>seu$cider_final_cluster</code>.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">seu</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/finalClustering.html">finalClustering</a></span><span class="op">(</span><span class="va">seu</span>, <span class="va">ider</span>, cutree.h <span class="op">=</span> <span class="fl">0.45</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">seu</span><span class="op">@</span><span class="va">meta.data</span><span class="op">)</span></span>
<span><span class="co">#&gt;                             orig.ident nCount_RNA nFeature_RNA Batch   Group</span></span>
<span><span class="co">#&gt; mouse1_lib1.final_cell_0001     mouse1      11003         2994 mouse    beta</span></span>
<span><span class="co">#&gt; mouse1_lib1.final_cell_0002     mouse1      10701         3565 mouse  ductal</span></span>
<span><span class="co">#&gt; mouse1_lib1.final_cell_0003     mouse1       9445         2401 mouse   delta</span></span>
<span><span class="co">#&gt; mouse1_lib1.final_cell_0004     mouse1       8193         2865 mouse schwann</span></span>
<span><span class="co">#&gt; mouse1_lib1.final_cell_0005     mouse1       7323         2355 mouse   delta</span></span>
<span><span class="co">#&gt; mouse1_lib1.final_cell_0006     mouse1       8504         2637 mouse    beta</span></span>
<span><span class="co">#&gt;                             Sample RNA_snn_res.0.8 seurat_clusters</span></span>
<span><span class="co">#&gt; mouse1_lib1.final_cell_0001 mouse1               9               9</span></span>
<span><span class="co">#&gt; mouse1_lib1.final_cell_0002 mouse1              13              13</span></span>
<span><span class="co">#&gt; mouse1_lib1.final_cell_0003 mouse1               7               7</span></span>
<span><span class="co">#&gt; mouse1_lib1.final_cell_0004 mouse1              21              21</span></span>
<span><span class="co">#&gt; mouse1_lib1.final_cell_0005 mouse1               7               7</span></span>
<span><span class="co">#&gt; mouse1_lib1.final_cell_0006 mouse1               8               8</span></span>
<span><span class="co">#&gt;                             initial_cluster CIDER_cluster</span></span>
<span><span class="co">#&gt; mouse1_lib1.final_cell_0001      beta_mouse             1</span></span>
<span><span class="co">#&gt; mouse1_lib1.final_cell_0002    ductal_mouse             2</span></span>
<span><span class="co">#&gt; mouse1_lib1.final_cell_0003     delta_mouse             3</span></span>
<span><span class="co">#&gt; mouse1_lib1.final_cell_0004   schwann_mouse             4</span></span>
<span><span class="co">#&gt; mouse1_lib1.final_cell_0005     delta_mouse             3</span></span>
<span><span class="co">#&gt; mouse1_lib1.final_cell_0006      beta_mouse             1</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="visualise-clustering-results">Visualise clustering results<a class="anchor" aria-label="anchor" href="#visualise-clustering-results"></a>
</h2>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">plot_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">plot_list</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/scatterPlot.html">scatterPlot</a></span><span class="op">(</span><span class="va">seu</span>, <span class="st">"tsne"</span>, colour.by <span class="op">=</span> <span class="st">"CIDER_cluster"</span>, title <span class="op">=</span> <span class="st">"asCIDER clustering results"</span><span class="op">)</span> </span>
<span><span class="va">plot_list</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/scatterPlot.html">scatterPlot</a></span><span class="op">(</span><span class="va">seu</span>, <span class="st">"tsne"</span>, colour.by <span class="op">=</span> <span class="st">"Group"</span>, title <span class="op">=</span> <span class="st">"Ground truth of cell populations"</span><span class="op">)</span> </span>
<span><span class="fu"><a href="https://wilkelab.org/cowplot/reference/plot_grid.html" class="external-link">plot_grid</a></span><span class="op">(</span>plotlist <span class="op">=</span> <span class="va">plot_list</span>, ncol <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<p><img src="asCIDER_files/figure-html/summary-tsne-plots-1.png" width="960"></p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">seu</span><span class="op">$</span><span class="va">CIDER_cluster</span>, <span class="va">seu</span><span class="op">$</span><span class="va">Batch</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">res</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>fill<span class="op">=</span><span class="va">Var2</span>, y<span class="op">=</span><span class="va">Freq</span>, x<span class="op">=</span><span class="va">Var1</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_bar</a></span><span class="op">(</span>position<span class="op">=</span><span class="st">"stack"</span>, stat<span class="op">=</span><span class="st">"identity"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">"CIDER_cluster"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">"Proportions"</span><span class="op">)</span></span></code></pre></div>
<p><img src="asCIDER_files/figure-html/unnamed-chunk-5-1.png" width="480"></p>
</div>
<div class="section level2">
<h2 id="reproducibility">Reproducibility<a class="anchor" aria-label="anchor" href="#reproducibility"></a>
</h2>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.4.1 (2024-06-14)</span></span>
<span><span class="co">#&gt; Platform: x86_64-apple-darwin20</span></span>
<span><span class="co">#&gt; Running under: macOS Monterey 12.5.1</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/lib/libRblas.0.dylib </span></span>
<span><span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: Europe/London</span></span>
<span><span class="co">#&gt; tzcode source: internal</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] viridis_0.6.5      viridisLite_0.4.2  cowplot_1.1.3      ggplot2_3.5.1     </span></span>
<span><span class="co">#&gt; [5] pheatmap_1.0.12    Seurat_5.1.0       SeuratObject_5.0.2 sp_2.1-4          </span></span>
<span><span class="co">#&gt; [9] CIDER_0.99.2      </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] RColorBrewer_1.1-3     rstudioapi_0.16.0      jsonlite_1.8.8        </span></span>
<span><span class="co">#&gt;   [4] magrittr_2.0.3         spatstat.utils_3.1-0   farver_2.1.2          </span></span>
<span><span class="co">#&gt;   [7] rmarkdown_2.27         fs_1.6.4               ragg_1.3.2            </span></span>
<span><span class="co">#&gt;  [10] vctrs_0.6.5            ROCR_1.0-11            spatstat.explore_3.3-2</span></span>
<span><span class="co">#&gt;  [13] htmltools_0.5.8.1      sass_0.4.9             sctransform_0.4.1     </span></span>
<span><span class="co">#&gt;  [16] parallelly_1.38.0      KernSmooth_2.23-24     bslib_0.7.0           </span></span>
<span><span class="co">#&gt;  [19] htmlwidgets_1.6.4      desc_1.4.3             ica_1.0-3             </span></span>
<span><span class="co">#&gt;  [22] plyr_1.8.9             plotly_4.10.4          zoo_1.8-12            </span></span>
<span><span class="co">#&gt;  [25] cachem_1.1.0           igraph_2.0.3           mime_0.12             </span></span>
<span><span class="co">#&gt;  [28] lifecycle_1.0.4        iterators_1.0.14       pkgconfig_2.0.3       </span></span>
<span><span class="co">#&gt;  [31] Matrix_1.7-0           R6_2.5.1               fastmap_1.2.0         </span></span>
<span><span class="co">#&gt;  [34] fitdistrplus_1.2-1     future_1.34.0          shiny_1.9.1           </span></span>
<span><span class="co">#&gt;  [37] digest_0.6.37          colorspace_2.1-1       patchwork_1.2.0       </span></span>
<span><span class="co">#&gt;  [40] tensor_1.5             RSpectra_0.16-2        irlba_2.3.5.1         </span></span>
<span><span class="co">#&gt;  [43] textshaping_0.4.0      labeling_0.4.3         progressr_0.14.0      </span></span>
<span><span class="co">#&gt;  [46] fansi_1.0.6            spatstat.sparse_3.1-0  httr_1.4.7            </span></span>
<span><span class="co">#&gt;  [49] polyclip_1.10-7        abind_1.4-5            compiler_4.4.1        </span></span>
<span><span class="co">#&gt;  [52] withr_3.0.1            doParallel_1.0.17      fastDummies_1.7.4     </span></span>
<span><span class="co">#&gt;  [55] highr_0.11             MASS_7.3-61            tools_4.4.1           </span></span>
<span><span class="co">#&gt;  [58] lmtest_0.9-40          httpuv_1.6.15          future.apply_1.11.2   </span></span>
<span><span class="co">#&gt;  [61] goftest_1.2-3          glue_1.7.0             dbscan_1.2.2          </span></span>
<span><span class="co">#&gt;  [64] nlme_3.1-165           promises_1.3.0         grid_4.4.1            </span></span>
<span><span class="co">#&gt;  [67] Rtsne_0.17             cluster_2.1.6          reshape2_1.4.4        </span></span>
<span><span class="co">#&gt;  [70] generics_0.1.3         gtable_0.3.5           spatstat.data_3.1-2   </span></span>
<span><span class="co">#&gt;  [73] tidyr_1.3.1            data.table_1.16.0      utf8_1.2.4            </span></span>
<span><span class="co">#&gt;  [76] spatstat.geom_3.3-2    RcppAnnoy_0.0.22       ggrepel_0.9.5         </span></span>
<span><span class="co">#&gt;  [79] RANN_2.6.2             foreach_1.5.2          pillar_1.9.0          </span></span>
<span><span class="co">#&gt;  [82] stringr_1.5.1          limma_3.60.6           spam_2.10-0           </span></span>
<span><span class="co">#&gt;  [85] RcppHNSW_0.6.0         later_1.3.2            splines_4.4.1         </span></span>
<span><span class="co">#&gt;  [88] dplyr_1.1.4            lattice_0.22-6         survival_3.7-0        </span></span>
<span><span class="co">#&gt;  [91] deldir_2.0-4           tidyselect_1.2.1       locfit_1.5-9.10       </span></span>
<span><span class="co">#&gt;  [94] miniUI_0.1.1.1         pbapply_1.7-2          knitr_1.48            </span></span>
<span><span class="co">#&gt;  [97] gridExtra_2.3          edgeR_4.2.2            scattermore_1.2       </span></span>
<span><span class="co">#&gt; [100] xfun_0.46              statmod_1.5.0          matrixStats_1.4.1     </span></span>
<span><span class="co">#&gt; [103] stringi_1.8.4          lazyeval_0.2.2         yaml_2.3.10           </span></span>
<span><span class="co">#&gt; [106] evaluate_0.24.0        codetools_0.2-20       kernlab_0.9-33        </span></span>
<span><span class="co">#&gt; [109] tibble_3.2.1           cli_3.6.3              uwot_0.2.2            </span></span>
<span><span class="co">#&gt; [112] xtable_1.8-4           reticulate_1.39.0      systemfonts_1.1.0     </span></span>
<span><span class="co">#&gt; [115] munsell_0.5.1          jquerylib_0.1.4        Rcpp_1.0.13           </span></span>
<span><span class="co">#&gt; [118] globals_0.16.3         spatstat.random_3.3-1  png_0.1-8             </span></span>
<span><span class="co">#&gt; [121] spatstat.univar_3.0-1  parallel_4.4.1         pkgdown_2.1.0         </span></span>
<span><span class="co">#&gt; [124] dotCall64_1.1-1        listenv_0.9.1          scales_1.3.0          </span></span>
<span><span class="co">#&gt; [127] ggridges_0.5.6         leiden_0.4.3.1         purrr_1.0.2           </span></span>
<span><span class="co">#&gt; [130] rlang_1.1.4</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<ol style="list-style-type: decimal">
<li>Baron, M. et al. A Single-Cell Transcriptomic Map of the Human and
Mouse Pancreas Reveals Inter- and Intra-cell Population Structure. Cell
Syst 3, 346–360.e4 (2016).</li>
<li>Satija R, et al. Spatial reconstruction of single-cell gene
expression data. Nature Biotechnology 33, 495-502 (2015).</li>
</ol>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Zhiyuan Hu, Christopher Yau, Ahmed Ahmed.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
