<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Getting Start with De Novo CIDER (dnCIDER): Cross-Species Pancreas Integration â€¢ CIDER</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Getting Start with De Novo CIDER (dnCIDER): Cross-Species Pancreas Integration">
<!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-207749729-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-207749729-1');
</script>
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">CIDER</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.2</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/asCIDER.html">Getting Start with Assisted CIDER (asCIDER)</a></li>
    <li><a class="dropdown-item" href="../articles/dnCIDER_highlevel.html">Getting Start with De Novo CIDER (dnCIDER): Cross-Species Pancreas Integration</a></li>
    <li><a class="dropdown-item" href="../articles/dnCIDER.html">Detailed walk-through of de novo CIDER (dnCIDER) on pancreas data</a></li>
    <li><a class="dropdown-item" href="../articles/evaluation.html">Evaluating scRNA-seq Integration Quality with CIDER</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/zhiyuan-hu-lab/CIDER/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Getting Start with De Novo CIDER (dnCIDER): Cross-Species Pancreas Integration</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/zhiyuan-hu-lab/CIDER/blob/HEAD/vignettes/dnCIDER_highlevel.Rmd" class="external-link"><code>vignettes/dnCIDER_highlevel.Rmd</code></a></small>
      <div class="d-none name"><code>dnCIDER_highlevel.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="what-is-dncider">What is dnCIDER?<a class="anchor" aria-label="anchor" href="#what-is-dncider"></a>
</h2>
<p><strong>dnCIDER</strong> is a computational method designed for
integrative analysis of single-cell RNA-seq data across batches or
species. This vignette demonstrates its application to a human-mouse
pancreas dataset, showing how to identify conserved cell populations
across species.</p>
</div>
<div class="section level2">
<h2 id="set-up">Set up<a class="anchor" aria-label="anchor" href="#set-up"></a>
</h2>
<p>In addition to <strong>CIDER</strong>, we will load the following
packages:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/zhiyuan-hu-lab/CIDER" class="external-link">CIDER</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Loading required package: SeuratObject</span></span>
<span><span class="co">#&gt; Loading required package: sp</span></span>
<span><span class="co">#&gt; 'SeuratObject' was built under R 4.4.0 but the current version is</span></span>
<span><span class="co">#&gt; 4.4.1; it is recomended that you reinstall 'SeuratObject' as the ABI</span></span>
<span><span class="co">#&gt; for R may have changed</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'SeuratObject'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     intersect, t</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">parallel</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://wilkelab.org/cowplot/" class="external-link">cowplot</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="load-pancreas-data">Load pancreas data<a class="anchor" aria-label="anchor" href="#load-pancreas-data"></a>
</h2>
<p>The example data can be downloaded from <a href="https://figshare.com/s/d5474749ca8c711cc205" class="external-link uri">https://figshare.com/s/d5474749ca8c711cc205</a>.</p>
<p>Pancreatic cell
data<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi></mi><mn>1</mn></msup><annotation encoding="application/x-tex">^1</annotation></semantics></math>
contain cells from human (8241 cells) and mouse (1886 cells).</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load counts matrix and metadata</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="st">"../data/pancreas_counts.RData"</span><span class="op">)</span> <span class="co"># count matrix</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="st">"../data/pancreas_meta.RData"</span><span class="op">)</span> <span class="co"># meta data/cell information</span></span>
<span></span>
<span><span class="co"># Create Seurat object</span></span>
<span><span class="va">seu</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://mojaveazure.github.io/seurat-object/reference/CreateSeuratObject.html" class="external-link">CreateSeuratObject</a></span><span class="op">(</span>counts <span class="op">=</span> <span class="va">pancreas_counts</span>, meta.data <span class="op">=</span> <span class="va">pancreas_meta</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Check batch composition</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">seu</span><span class="op">$</span><span class="va">Batch</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; human mouse </span></span>
<span><span class="co">#&gt;  8241  1886</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="perform-dncider-high-level">Perform dnCIDER (high-level)<a class="anchor" aria-label="anchor" href="#perform-dncider-high-level"></a>
</h2>
<p>DnCIDER contains three steps.</p>
<div class="section level3">
<h3 id="step-1-initial-clustering">Step 1: Initial Clustering<a class="anchor" aria-label="anchor" href="#step-1-initial-clustering"></a>
</h3>
<p>Performs preprocessing and generates initial clusters within each
batch:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">seu</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/initialClustering.html">initialClustering</a></span><span class="op">(</span></span>
<span>  <span class="va">seu</span>,</span>
<span>  additional.vars.to.regress <span class="op">=</span> <span class="st">"Sample"</span>,  <span class="co"># Regress out sample-specific effects</span></span>
<span>  dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">15</span>                             <span class="co"># PCA dimensions to use</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt;   |                                                                              |                                                                      |   0%  |                                                                              |===================================                                   |  50%  |                                                                              |======================================================================| 100%</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="step-2-compute-ider">Step 2: Compute IDER<a class="anchor" aria-label="anchor" href="#step-2-compute-ider"></a>
</h3>
<p>Estimates batch-corrected similarity matrices:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ider</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getIDEr.html">getIDEr</a></span><span class="op">(</span></span>
<span>  <span class="va">seu</span>,</span>
<span>  downsampling.size <span class="op">=</span> <span class="fl">35</span>,   <span class="co"># Cells per cluster for downsampling</span></span>
<span>  use.parallel <span class="op">=</span> <span class="cn">FALSE</span>,     <span class="co"># Disable parallelization for reproducibility</span></span>
<span>  verbose <span class="op">=</span> <span class="cn">FALSE</span>           <span class="co"># Suppress progress messages</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="step-3-final-integrated-clustering">Step 3: Final Integrated Clustering<a class="anchor" aria-label="anchor" href="#step-3-final-integrated-clustering"></a>
</h3>
<p>Merges clusters using IDER-derived similarities:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">seu</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/finalClustering.html">finalClustering</a></span><span class="op">(</span></span>
<span>  <span class="va">seu</span>,</span>
<span>  <span class="va">ider</span>,</span>
<span>  cutree.h <span class="op">=</span> <span class="fl">0.35</span>  <span class="co"># Height for hierarchical clustering cut</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="visualise-clustering-results">Visualise clustering results<a class="anchor" aria-label="anchor" href="#visualise-clustering-results"></a>
</h2>
<p>We use the Seurat pipeline to perform normalisation
(<code>NormalizeData</code>), preprocessing
(<code>FindVariableFeatures</code> and <code>ScaleData</code>) and
dimension reduction (<code>RunPCA</code> and <code>RunTSNE</code>).</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Preprocessing for Visualization</span></span>
<span><span class="va">seu</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/NormalizeData.html" class="external-link">NormalizeData</a></span><span class="op">(</span><span class="va">seu</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">seu</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindVariableFeatures.html" class="external-link">FindVariableFeatures</a></span><span class="op">(</span><span class="va">seu</span>, selection.method <span class="op">=</span> <span class="st">"vst"</span>, nfeatures <span class="op">=</span> <span class="fl">2000</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">seu</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/ScaleData.html" class="external-link">ScaleData</a></span><span class="op">(</span><span class="va">seu</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">seu</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/RunPCA.html" class="external-link">RunPCA</a></span><span class="op">(</span><span class="va">seu</span>, npcs <span class="op">=</span> <span class="fl">20</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">seu</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/RunTSNE.html" class="external-link">RunTSNE</a></span><span class="op">(</span><span class="va">seu</span>, reduction <span class="op">=</span> <span class="st">"pca"</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">12</span><span class="op">)</span></span></code></pre></div>
<p>Next we plot integrated clusters vs.Â ground truth. By comparing the
dnCIDER results to the cell annotation from the
publication<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi></mi><mn>1</mn></msup><annotation encoding="application/x-tex">^1</annotation></semantics></math>,
we observe that dnCIDER correctly identify the majority of populations
across two species.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Generate plots</span></span>
<span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/scatterPlot.html">scatterPlot</a></span><span class="op">(</span><span class="va">seu</span>, <span class="st">"tsne"</span>, </span>
<span>                  colour.by <span class="op">=</span> <span class="st">"CIDER_cluster"</span>, </span>
<span>                  title <span class="op">=</span> <span class="st">"Integrated Clusters (dnCIDER)"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/scatterPlot.html">scatterPlot</a></span><span class="op">(</span><span class="va">seu</span>, <span class="st">"tsne"</span>, </span>
<span>                  colour.by <span class="op">=</span> <span class="st">"Group"</span>, </span>
<span>                  title <span class="op">=</span> <span class="st">"Original Cell Types"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Arrange side-by-side</span></span>
<span><span class="fu"><a href="https://wilkelab.org/cowplot/reference/plot_grid.html" class="external-link">plot_grid</a></span><span class="op">(</span><span class="va">p1</span>, <span class="va">p2</span>, ncol <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<p><img src="dnCIDER_highlevel_files/figure-html/tsne-plot-CIDER-results-1.png" width="768"></p>
<p>Interpretation: dnCIDER successfully aligns human (prefix h) and
mouse (m) cell types. For example:</p>
<ul>
<li>Beta cells (hBeta/mBeta) form a unified cluster</li>
<li>Alpha cells (hAlpha/mAlpha) show cross-species alignment</li>
<li>Minor populations like Acinar and Ductal are conserved</li>
</ul>
</div>
<div class="section level2">
<h2 id="notes-best-practices">Notes &amp; Best Practices<a class="anchor" aria-label="anchor" href="#notes-best-practices"></a>
</h2>
<ol style="list-style-type: decimal">
<li>
<strong>Downsampling Size</strong>: Adjust
<code>downsampling.size</code> (default: 35) if clusters are small.</li>
<li>
<strong>Batch Variable</strong>: Ensure your Seurat object contains
a batch identifier (default column name: <code>"Batch"</code>).</li>
<li>
<strong>Visualization</strong>: Always validate integration using
known marker genes in addition to embeddings.</li>
<li>
<strong>Runtime</strong>: For large datasets, enable parallelization
with <code>use.parallel = TRUE</code>.</li>
</ol>
</div>
<div class="section level2">
<h2 id="reproducibility">Reproducibility<a class="anchor" aria-label="anchor" href="#reproducibility"></a>
</h2>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.4.1 (2024-06-14)</span></span>
<span><span class="co">#&gt; Platform: x86_64-apple-darwin20</span></span>
<span><span class="co">#&gt; Running under: macOS Monterey 12.5.1</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/lib/libRblas.0.dylib </span></span>
<span><span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: Europe/London</span></span>
<span><span class="co">#&gt; tzcode source: internal</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] parallel  stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] cowplot_1.1.3      Seurat_5.1.0       SeuratObject_5.0.2 sp_2.1-4          </span></span>
<span><span class="co">#&gt; [5] CIDER_0.99.2      </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] RColorBrewer_1.1-3     rstudioapi_0.16.0      jsonlite_1.8.8        </span></span>
<span><span class="co">#&gt;   [4] magrittr_2.0.3         spatstat.utils_3.1-0   farver_2.1.2          </span></span>
<span><span class="co">#&gt;   [7] rmarkdown_2.27         fs_1.6.4               ragg_1.3.2            </span></span>
<span><span class="co">#&gt;  [10] vctrs_0.6.5            ROCR_1.0-11            spatstat.explore_3.3-2</span></span>
<span><span class="co">#&gt;  [13] htmltools_0.5.8.1      sass_0.4.9             sctransform_0.4.1     </span></span>
<span><span class="co">#&gt;  [16] parallelly_1.38.0      KernSmooth_2.23-24     bslib_0.7.0           </span></span>
<span><span class="co">#&gt;  [19] htmlwidgets_1.6.4      desc_1.4.3             ica_1.0-3             </span></span>
<span><span class="co">#&gt;  [22] plyr_1.8.9             plotly_4.10.4          zoo_1.8-12            </span></span>
<span><span class="co">#&gt;  [25] cachem_1.1.0           igraph_2.0.3           mime_0.12             </span></span>
<span><span class="co">#&gt;  [28] lifecycle_1.0.4        iterators_1.0.14       pkgconfig_2.0.3       </span></span>
<span><span class="co">#&gt;  [31] Matrix_1.7-0           R6_2.5.1               fastmap_1.2.0         </span></span>
<span><span class="co">#&gt;  [34] fitdistrplus_1.2-1     future_1.34.0          shiny_1.9.1           </span></span>
<span><span class="co">#&gt;  [37] digest_0.6.37          colorspace_2.1-1       patchwork_1.2.0       </span></span>
<span><span class="co">#&gt;  [40] tensor_1.5             RSpectra_0.16-2        irlba_2.3.5.1         </span></span>
<span><span class="co">#&gt;  [43] textshaping_0.4.0      labeling_0.4.3         progressr_0.14.0      </span></span>
<span><span class="co">#&gt;  [46] fansi_1.0.6            spatstat.sparse_3.1-0  httr_1.4.7            </span></span>
<span><span class="co">#&gt;  [49] polyclip_1.10-7        abind_1.4-5            compiler_4.4.1        </span></span>
<span><span class="co">#&gt;  [52] withr_3.0.1            doParallel_1.0.17      viridis_0.6.5         </span></span>
<span><span class="co">#&gt;  [55] fastDummies_1.7.4      highr_0.11             MASS_7.3-61           </span></span>
<span><span class="co">#&gt;  [58] tools_4.4.1            lmtest_0.9-40          httpuv_1.6.15         </span></span>
<span><span class="co">#&gt;  [61] future.apply_1.11.2    goftest_1.2-3          glue_1.7.0            </span></span>
<span><span class="co">#&gt;  [64] dbscan_1.2.2           nlme_3.1-165           promises_1.3.0        </span></span>
<span><span class="co">#&gt;  [67] grid_4.4.1             Rtsne_0.17             cluster_2.1.6         </span></span>
<span><span class="co">#&gt;  [70] reshape2_1.4.4         generics_0.1.3         gtable_0.3.5          </span></span>
<span><span class="co">#&gt;  [73] spatstat.data_3.1-2    tidyr_1.3.1            data.table_1.16.0     </span></span>
<span><span class="co">#&gt;  [76] utf8_1.2.4             spatstat.geom_3.3-2    RcppAnnoy_0.0.22      </span></span>
<span><span class="co">#&gt;  [79] ggrepel_0.9.5          RANN_2.6.2             foreach_1.5.2         </span></span>
<span><span class="co">#&gt;  [82] pillar_1.9.0           stringr_1.5.1          limma_3.60.6          </span></span>
<span><span class="co">#&gt;  [85] spam_2.10-0            RcppHNSW_0.6.0         later_1.3.2           </span></span>
<span><span class="co">#&gt;  [88] splines_4.4.1          dplyr_1.1.4            lattice_0.22-6        </span></span>
<span><span class="co">#&gt;  [91] survival_3.7-0         deldir_2.0-4           tidyselect_1.2.1      </span></span>
<span><span class="co">#&gt;  [94] locfit_1.5-9.10        miniUI_0.1.1.1         pbapply_1.7-2         </span></span>
<span><span class="co">#&gt;  [97] knitr_1.48             gridExtra_2.3          edgeR_4.2.2           </span></span>
<span><span class="co">#&gt; [100] scattermore_1.2        xfun_0.46              statmod_1.5.0         </span></span>
<span><span class="co">#&gt; [103] matrixStats_1.4.1      pheatmap_1.0.12        stringi_1.8.4         </span></span>
<span><span class="co">#&gt; [106] lazyeval_0.2.2         yaml_2.3.10            evaluate_0.24.0       </span></span>
<span><span class="co">#&gt; [109] codetools_0.2-20       kernlab_0.9-33         tibble_3.2.1          </span></span>
<span><span class="co">#&gt; [112] cli_3.6.3              uwot_0.2.2             xtable_1.8-4          </span></span>
<span><span class="co">#&gt; [115] reticulate_1.39.0      systemfonts_1.1.0      munsell_0.5.1         </span></span>
<span><span class="co">#&gt; [118] jquerylib_0.1.4        Rcpp_1.0.13            globals_0.16.3        </span></span>
<span><span class="co">#&gt; [121] spatstat.random_3.3-1  png_0.1-8              spatstat.univar_3.0-1 </span></span>
<span><span class="co">#&gt; [124] pkgdown_2.1.0          ggplot2_3.5.1          dotCall64_1.1-1       </span></span>
<span><span class="co">#&gt; [127] listenv_0.9.1          viridisLite_0.4.2      scales_1.3.0          </span></span>
<span><span class="co">#&gt; [130] ggridges_0.5.6         leiden_0.4.3.1         purrr_1.0.2           </span></span>
<span><span class="co">#&gt; [133] rlang_1.1.4</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<ol style="list-style-type: decimal">
<li>Baron, M. et al.Â A Single-Cell Transcriptomic Map of the Human and
Mouse Pancreas Reveals Inter- and Intra-cell Population Structure. Cell
Syst 3, 346â€“360.e4 (2016).</li>
<li>Satija R, et al.Â Spatial reconstruction of single-cell gene
expression data. Nature Biotechnology 33, 495-502 (2015).</li>
</ol>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Zhiyuan Hu, Christopher Yau, Ahmed Ahmed.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
